### PR TITLE
OpenPCDet-EPNet

### PR DESCRIPTION
**Pull request by wyd9697**
@zbaishancha Hello, thank you very much for your open-source work. I would like to use the OpenPCDet framework based EPNet to test multiple categories, but it has been unsuccessful during the installation process. I would like to inquire about the Python, Pytorch, and Cuda versions you are using. Thank you again.

### CODE REVIEW

**Comment by zbaishancha** [comment]
@wyd9697 Hi, my version is as follows：python3.8
image

Also, I don't think there are any special requirements for the version of torch, just the ability to run openpcdet successfully.

I wish you success!

**Comment by wyd9697** [comment]
Thank you for your reply, it may be a problem with our cuda version, after adjusting it, it has been successfully installed. Your work has helped us a lot and we have learned a lot, thanks again. In addition, I got a reply from the original author of EPNet, according to his parameter settings, the algorithm performance on OPENPCDet is close to the original paper, and I send you a copy for your reference.


| |
wangyadong
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 6/19/2023 10:07 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [open-mmlab/OpenPCDet] OpenPCDet-EPNet (PR #1382) |

@wyd9697 Hi, my version is as follows：python3.8


Also, I don't think there are any special requirements for the version of torch, just the ability to run openpcdet successfully.

I wish you success!

**Comment by zbaishancha** [comment]
@wyd9697 You are welcome. In addition, I migrated EPNet, in which CE_Loss was not successfully migrated in the second stage, only the backbone of EPNet was migrated and ce loss was used in the first stage. then the rest of the configuration parameters used the network parameters of PointRCNN, and the final training results were obtained as follows:
image
How do you handle celoss? And how are the detailed training parameters set? Thank you very much if you can share!

**Comment by zbaishancha** [comment]
@wyd9697 Also, I recently finished migrating epnet++, so if you have any questions, feel free to discuss them with me!

**Comment by wyd9697** [comment]
Hello, the original author of the OPENPCDet framework also seems to be using the original loss of PointRCNN. Because the framework I deployed is 0.6 and the original author's code is 0.3, I have modified the parameters based on your code, and the pedestrian and bicycle accuracy after training is slightly lower than the original paper.




| |
wangyadong
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 6/19/2023 10:26 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [open-mmlab/OpenPCDet] OpenPCDet-EPNet (PR #1382) |

@wyd9697 You are welcome. In addition, I migrated EPNet, in which CE_Loss was not successfully migrated in the second stage, only the backbone of EPNet was migrated and ce loss was used in the first stage. then the rest of the configuration parameters used the network parameters of PointRCNN, and the final training results were obtained as follows:

How do you handle celoss? And how are the detailed training parameters set? Thank you very much if you can share!

**Comment by zbaishancha** [comment]
@wyd9697 Thank you very much for your work, could you please share your code? Also, it's been a long time since I've submitted my repository, and there are some places where I've made changes. When I have some free time, I'll reorganize it and submit the code for epnet++ as well.

**Comment by wyd9697** [comment]
Hello, I sent the code and it was returned, saying that it exceeded the recipient's capacity. I sent the modified part this time, and the rest of the code is the same as yours.


| |
wangyadong
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 6/19/2023 10:57 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [open-mmlab/OpenPCDet] OpenPCDet-EPNet (PR #1382) |

@wyd9697 Thank you very much for your work, could you please share your code? Also, it's been a long time since I've submitted my repository, and there are some places where I've made changes. When I have some free time, I'll reorganize it and submit the code for epnet++ as well.

**Comment by zbaishancha** [comment]
@wyd9697  Thank you very much, you can send the modified code to my Email z_baishancha@126.com. Also, my QQ number is 2092464036, welcome to contact.

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @wyd9697,

We'd like to express our appreciation for your valuable contributions to the OpenPCDet. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!
